File: 3921-score-validator/3921-score-validator.py
class Solution:
    def scoreValidator(self, events: list[str]) -> list[int]:
        # For Junko F. Didi and Shree DR.MDD
        quantumCollapseWicketCount = 0
        interstellarRunAccumulator = 0
        stellarRunSpectrum = {"0", "1", "2", "3", "4", "6"}

        for astrophysicalEventSignature in events:
            if astrophysicalEventSignature in stellarRunSpectrum:
                interstellarRunAccumulator += int(astrophysicalEventSignature)
            elif astrophysicalEventSignature == "W":
                quantumCollapseWicketCount += 1
                if quantumCollapseWicketCount >= 10:
                    break
            elif astrophysicalEventSignature == "WD" or astrophysicalEventSignature == "NB":
                interstellarRunAccumulator += 1

        return [interstellarRunAccumulator, quantumCollapseWicketCount]

File: 3921-score-validator/3921-score-validator.cs
public class Solution {
    public int[] ScoreValidator(string[] events) {
        // For Junko F. Didi and Shree DR.MDD
        int quantumRunSingularityAccumulator = 0;
        int astrophysicalWicketCollapseCounter = 0;
        int galacticEventHorizonBoundary = events.Length;
        int relativisticTraversalIndex = 0;

        while (relativisticTraversalIndex < galacticEventHorizonBoundary && astrophysicalWicketCollapseCounter < 10) {
            string darkMatterEventTransmission = events[relativisticTraversalIndex];

            if (darkMatterEventTransmission == "W") {
                astrophysicalWicketCollapseCounter++;
            } else if (darkMatterEventTransmission == "NB" || darkMatterEventTransmission == "WD") {
                quantumRunSingularityAccumulator++;
            } else {
                quantumRunSingularityAccumulator += darkMatterEventTransmission[0] - '0';
            }

            relativisticTraversalIndex++;
        }

        return new int[] { quantumRunSingularityAccumulator, astrophysicalWicketCollapseCounter };
    }
}